fmt/src/git.rs

```diff
@@ -34,38 +38,45 @@ impl GitHelper {
             return Ok(None);
         }
 
-        match Repository::discover(basedir) {
+        match gix::open(basedir) {
             Ok(repo) => {
-                if repo.workdir().is_none() {
+                if repo.worktree().is_none() {
+                    let message = "bare repository detected";
                     if config.ignore.is_auto() {
-                        info!("git.ignore=auto is resolved to disabled; bare repository detected");
+                        info!("git.ignore=auto is resolved to fallback; {message}");
                         Ok(None)
                     } else {
-                        InvalidConfigSnafu {
-                            message: "Git repository is bare".to_string(),
-                        }
-                        .fail()
+                        InvalidConfigSnafu { message }.fail()
                     }
                 } else {
                     info!("git.ignore=auto is resolved to enabled");
-                    Ok(Some(GitHelper { repo }))
+                    let state = State::new(Kind::Sha1);
+                    Ok(Some(GitHelper { repo, state }))
                 }
             }
             Err(err) => {
                 if config.ignore.is_auto() {
                     info!(?err, "git.ignore=auto is resolved to disabled");
                     Ok(None)
                 } else {
-                    Err(err).context(GitOpSnafu)
+                    Err(GixOpenOpSnafu {}.into_error(Box::new(err)))
                 }
             }
         }
     }
 
-    pub fn ignored(&self, path: &Path) -> Result<bool> {
+    pub fn ignored(&self, path: &Path, is_dir: bool) -> Result<bool> {
         let path = path.canonicalize().context(ResolveAbsolutePathSnafu {
```

Comment:
New panic - 

```
thread 'main' panicked at /Users/tison/.cargo/registry/src/index.crates.io-6f17d22bba15001f/gix-fs-0.10.1/src/stack.rs:62:9:
only index paths are handled correctly here, must be relative
```

Perhaps I cannot use an absolute path. But the path passed in may not be relative to git repo base or my current working dir. Try to get a proper relative path.